packages/market-service/src/idle/idle.ts

```diff
@@ -0,0 +1,89 @@
+import { fromAssetId, toAssetId } from '@shapeshiftoss/caip'
+import { ethereum } from '@shapeshiftoss/chain-adapters'
+import { IdleInvestor } from '@shapeshiftoss/investor-idle'
+import { MarketCapResult, MarketData, MarketDataArgs } from '@shapeshiftoss/types'
+import * as unchained from '@shapeshiftoss/unchained-client'
+
+import { MarketService } from '../api'
+import { ProviderUrls } from '../market-service-manager'
+
+export class IdleMarketService implements MarketService {
+  baseUrl = ''
+  providerUrls: ProviderUrls
+  idleInvestor: IdleInvestor
+
+  constructor({ providerUrls }: { providerUrls: ProviderUrls }) {
+    this.providerUrls = providerUrls
+    this.idleInvestor = new IdleInvestor({
+      chainAdapter: new ethereum.ChainAdapter({
+        providers: {
+          ws: new unchained.ws.Client<unchained.ethereum.Tx>(
+            this.providerUrls.unchainedEthereumWsUrl,
+          ),
+          http: new unchained.ethereum.V1Api(
+            new unchained.ethereum.Configuration({
+              basePath: this.providerUrls.unchainedEthereumHttpUrl,
+            }),
+          ),
+        },
+        rpcUrl: this.providerUrls.jsonRpcProviderUrl,
+      }),
+      providerUrl: this.providerUrls.jsonRpcProviderUrl,
+    })
+    ;(async () => {
+      await this.idleInvestor.initialize()
+    })()
+  }
+
+  async findAll() {
+    const idleOpportunities = await (async () => {
+      const maybeOpportunities = await this.idleInvestor.findAll()
+      if (maybeOpportunities.length) return maybeOpportunities
+
+      await this.idleInvestor.initialize()
```

Comment:
prob don't need to reinitialize if we do this in the constructor

Reply:
Added a comment WRT the unsafety of the constructor initialization https://github.com/shapeshift/lib/pull/1097/files#r1029517425 - happy to remove it in the constructor since it looks like we are too fast to finish initializing before these are run and always end up re-initializing